app/src/main/java/org/greenstand/android/TreeTracker/viewmodels/ConfigViewModel.kt

```diff
@@ -0,0 +1,29 @@
+package org.greenstand.android.TreeTracker.viewmodels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import org.greenstand.android.TreeTracker.models.Configuration
+import org.greenstand.android.TreeTracker.models.LocationDataConfig
+import org.greenstand.android.TreeTracker.models.LocationUpdateManager
+
+class ConfigViewModel(
+    private val configuration: Configuration,
+    private val locationUpdateManager: LocationUpdateManager
+) : ViewModel() {
+
+    private val locationConfigLiveData = MutableLiveData<LocationDataConfig>()
+
+    fun getLocationDataConfig(): LiveData<LocationDataConfig> {
+        locationConfigLiveData.value = configuration.locationDataConfig

```

Comment:
Looks like this can be replace with a call to 'updateLocationDataConfig()'

Reply:
Sorry, I don't understand your observation here.

getLocationDataConfig returns LiveData with most current value for locationDataConfig and used by UI code interested in the changed value.  We would use updateLocationDataConfig only to update the locationDataConfig in Configuration and to persist the changes.